app/src/main/java/com/example/habbit/activities/AllHabitsActivity.java

```diff
@@ -115,7 +115,8 @@ public void onClick(View v) {
                     Habit habit = new Habit(Objects.requireNonNull(habitData.get("title")).toString(),
                             Objects.requireNonNull(habitData.get("reason")).toString(),
                             Objects.requireNonNull(habitData.get("date")).toString(),
-                            (HashMap<String, Boolean>) Objects.requireNonNull(habitData.get("schedule")));
+                            (HashMap<String, Boolean>) Objects.requireNonNull(habitData.get("schedule")),
+                            (boolean) Objects.requireNonNull(habitData.get("publicity")));
```

Comment:
publicity is probably not the best name for this. How about isPublic or isPrivate?